### DISCUSSION TITLE
Proper way of handling a float variable.

### DISCUSSION THREAD

**Discussion by PapitoPapito** [KyryloKuzyk/PrimeTween#113] [FAQ] [answered]
Hi.

 Thanks for sharing your tween system. 
I use custom tween with inspector integration to tween a float variable as follow:

  void StartTween()
    { Tween.Custom(floatTweenSettings, onValueChange: newVal => floatVariable.Value = newVal); }

 The float variable is a scriptable object value.

 Is it the proper way to use it?
Also, I was wondering as it looked like a event system, should I unsubscribe somewhere to avoid redundance?
 Lastly, does the tween remains somewhere? Do I need to kill it like in Dotween?

 Thanks for your help

**Comment by KyryloKuzyk** [comment, marked as answer]
Hey!
1. Yes, this is the correct way to do custom animations. You can also improve it slightly by removing the memory allocation inside the delegate: https://github.com/KyryloKuzyk/PrimeTween?tab=readme-ov-file#zero-allocations-with-delegates
2. No, you don't have to unsubscribe from custom animations. The delegate is cleared automatically upon animation completion.
3. The tween is stored in the list inside PrimeTweenManager and you debug all currently running tween if you select this object in the Inspector: https://github.com/KyryloKuzyk/PrimeTween?tab=readme-ov-file#debugging-tweens
You don't have to Kill() tweens in PrimeTween, they are removed from the list automatically upon completion. You can even destroy the tween's target while the tween is still running: https://github.com/KyryloKuzyk/PrimeTween/discussions/4

**Comment by PapitoPapito** [comment]
Thanks!
